### PR TITLE
2402: Update metro dependencies

### PR DESCRIPTION
**Pull request by sarahsporck**
### Short description

Update metro dependencies.

### Proposed changes

- remove metro resolver as it is now unused
- use the react-native metro config as base
- use extraNodeModules

### Side effects

- A warning is displayed that some invalid settings are set in the metro config. These should disappear once we update to RN 72 (although this is not a peer dep of the package)
- Rendering SVGs

### Resolved issues

Fixes: #2402

---

### CODE REVIEW

File: native/metro.config.js [hunk 1]
```diff
@@ -6,55 +6,45 @@
...
     },
-    watchFolders: [path.resolve(__dirname, '../')],
-    transformer: {
-      allowOptionalDependencies: true,
-      babelTransformerPath: require.resolve('react-native-svg-transformer'),
-      getTransformOptions: async () => ({
-        transform: {
-          experimentalImportSupport: false,
-          inlineRequires: true,
-        },
```
Comment (sarahsporck) [inline native/metro.config.js:57]: This is set by react native already:
https://github.com/facebook/react-native/blob/main/packages/metro-config/index.js#L70-L82

**Review by LeandraH — APPROVED** [review]
Nice! Tested on real iOS

**Comment by steffenkleinle** [comment]
> **Side effects**
>    - Rendering SVGs

What does that mean? Could you elaborate please?

**Review by steffenkleinle — APPROVED** [review]

File: native/metro.config.js [hunk 2]
```diff
@@ -6,55 +6,45 @@
...
-              const buildConfigName = process.env.BUILD_CONFIG_NAME || 'integreat-test-cms'
-              // Proxy the (non-existing) module 'build-config-name' to the name of the right build config.
-              // Passing environment variables is not possible without either babel or post processing otherwise.
-              return path.resolve(__dirname, '../build-configs', buildConfigName, 'build-config-name/index.ts')
-            }
-
-            return path.resolve(__dirname, `node_modules/${name}`)
-          },
-        },
+/** @type {import('metro-config').MetroConfig} */
```
[the remark this answers is not in the document]
Comment (sarahsporck) [reply native/metro.config.js:18]: Not that I know of. We could leave the first type import, as we actually don't really use it anyway, but as these [doc strings](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type) are only "executed" during development as this is only executed for the linter, I guess.

File: native/metro.config.js [hunk 3]
```diff
@@ -6,55 +6,45 @@
  *
```
[the remark this answers is not in the document]
Comment (sarahsporck) [reply native/metro.config.js:6]: Will remove. I think the react-native upgrade helper added this and we just copy pasted it here ^^

**Comment by sarahsporck** [comment]
> > **Side effects**
> > 
> > * Rendering SVGs
> 
> What does that mean? Could you elaborate please?

Our svg renderer depends is basically configured in our metro.config.js. If I messed sth. up the svgs would not appear, so I think there is no issue

**Pull request merged**